lib/srv/desktop/discovery.go

```diff
@@ -414,12 +422,17 @@ func (s *WindowsService) startDynamicReconciler(ctx context.Context) (*services.
 	go func() {
 		defer s.cfg.Logger.DebugContext(ctx, "DynamicWindowsDesktop resource watcher done.")
 		defer watcher.Close()
+		tickDuration := 5 * time.Minute
+		expiryDuration := tickDuration + 10*time.Second
```

Comment:
As written, we are having a small race whenever we need to update the desktop expiry time. On each tick we only have 10 seconds of expiry time remaining, which feels very short and may eventually lead to intermittent availability issues.

I think we can simply raise the delta without significant problems:

```suggestion
		tickDuration := 5 * time.Minute
		expiryDuration := tickDuration + 2*time.Minute
```

However, if there are thousands of desktops within the cluster, this will still add nontrivial load to the backend.

What is the highest we can go here without sacrificing UX?

Reply:
I bumped the expiry. We do similar thing for desktops from config file and discovered through LDAP (we refresh them every 5 min) so I think we should be OK here, @zmb3 thoughts?